src/common/storage/util.js

```diff
@@ -56,6 +199,76 @@ define(['common/storage/constants', 'common/util/jsonPatcher'], function (CONSTA
             return !!(coreObj.oldHash && coreObj.newHash && coreObj.oldData && coreObj.newData);
         },
         getChangedNodes: jsonPatcher.getChangedNodes,
-        applyPatch: jsonPatcher.apply
+        applyPatch: jsonPatcher.apply,
+
+        /**
+         * Extracts a serializable json representation of a project tree.
+         * To specify starting point set one of the four options. If more than one is set the order of precedence is:
+         * rootHash, commitHash, tagName and branchName.
+         *
+         * @param {ProjectInterface} project
+         * @param {object} parameters - Specifies which project tree should be serialized:
+         * @param {string} [parameters.rootHash] - The hash of the tree root.
+         * @param {string} [parameters.commitHash] - The tree associated with the commitHash.
+         * @param {string} [parameters.tagName] - The tree at the given tag.
+         * @param {string} [parameters.branchName] - The tree at the given branch.
+         * @param {function} callback
+         */
+        getProjectJson: function (project, parameters, callback) {
+            var deferred = Q.defer(),
+                rawJson;
+
+            _getRootHash(project, parameters)
```

Comment:
`parameters || {}` - that way it won't be type error but rather your descriptive exception.